### PR TITLE
Apps: central added

### PR DESCRIPTION
**Pull request by KKopyscinski**
Added central application which presents how to connect to peripheral.

### CODE REVIEW

**Comment by sjanc** [comment]
some general comments on how I'd like to see this sample:

- no need to parse full adv data and print those,  just look for specified UUID and print info when it is found
- then initiate connection
- when connected (or failed) print proper info about  that
- update conn params, notify when updated
- change PHY eg to 2M and notify when done,
- disconnect
- wait for few sconds and go back to 1

File: apps/central/src/main.c [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                                             BLE_GAP_LE_PHY_2M_MASK,
+                                             BLE_GAP_LE_PHY_2M_MASK,
+                                             BLE_GAP_LE_PHY_CODED_ANY);
+            if (rc == 0) {
+                MODLOG_DFLT(INFO, "rx and tx PHYs set to 2M\n");
+            } else {
+                MODLOG_DFLT(INFO, "failed to set rx and tx PHYs\n");
+            }
+            ble_gap_terminate(event->connect.conn_handle, 0x13);
+            os_time_ms_to_ticks(delay_ms, &ticks);
```
Comment (rymanluk) [inline apps/central/src/main.c:70]: os_time_delay(os_time_ms_to_ticks32(2000)); ?
Also please write comment why it is here.

File: apps/central/src/main.c [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            rc = ble_gap_set_prefered_le_phy(event->connect.conn_handle,
+                                             BLE_GAP_LE_PHY_2M_MASK,
+                                             BLE_GAP_LE_PHY_2M_MASK,
+                                             BLE_GAP_LE_PHY_CODED_ANY);
+            if (rc == 0) {
+                MODLOG_DFLT(INFO, "rx and tx PHYs set to 2M\n");
+            } else {
+                MODLOG_DFLT(INFO, "failed to set rx and tx PHYs\n");
+            }
+            ble_gap_terminate(event->connect.conn_handle, 0x13);
```
[the remark this answers is not in the document]
Comment (Reynevan94) [reply apps/central/src/main.c:69]: As far as I understand, this GAP event should be handled by peripheral and on central side we don't have easy way to find out if it completed, but if I'm mistaken, I'm happy to change that.

[the remark this answers is not in the document]
Comment (rymanluk) [reply apps/central/src/main.c:71]: Event is sent on  both sides.

[the remark this answers is not in the document]
Comment (Reynevan94) [reply apps/central/src/main.c]: I added this feature, but I couldn't get this event called. Maybe other side is configured in a way this change can't be made, so this should be checked.

File: apps/central/src/main.c (hunk 1 above)
Comment (rymanluk) [reply apps/central/src/main.c:71]: this seems to be unresolved.

Comment (Reynevan94) [reply apps/central/src/main.c]: That was simply overlooked and it's fixed now. os_time_delay(os_time_ms_to_ticks32(delay_ms)); works perfect

File: apps/central/src/main.c [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+#include "os/os.h"
+#include "console/console.h"
+#include "host/ble_hs.h"
+#include "host/util/util.h"
+#include "console/console.h"
+#include "log/log.h"
+
+static uint8_t g_own_addr_type;
+static uint32_t delay_ms = 2000;
+static os_time_t ticks;
```
Comment (rymanluk) [inline apps/central/src/main.c:30]: this is not needed anymore.

File: apps/central/src/main.c [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+#include "sysinit/sysinit.h"
+#include "os/os.h"
+#include "console/console.h"
+#include "host/ble_hs.h"
+#include "host/util/util.h"
+#include "console/console.h"
+#include "log/log.h"
+
+static uint8_t g_own_addr_type;
+static uint32_t delay_ms = 2000;
```
Comment (rymanluk) [inline apps/central/src/main.c:29]: this delay is also not needed here - you ca put is as value where it is used

Comment (KKopyscinski) [reply apps/central/src/main.c]: I agree with both - it's fixed now

File: apps/central/src/main.c [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+                                             BLE_GAP_LE_PHY_CODED_ANY);
+            if (rc == 0) {
+                MODLOG_DFLT(INFO,"PHY update successful\n");
+            } else {
+                MODLOG_DFLT(INFO,"PHY update failed\n");
+            }
+            /* Before calling scan() again, wait predefined time delay_ms.
+               This value has to be converted to ticks first before calling
+               os_time_delay() */
+            os_time_delay(os_time_ms_to_ticks32(2000));
```
Comment (sjanc) [inline apps/central/src/main.c:72]: you are blocking host task with that, you should not do that

I think for central app we can skip phy updates for now and just have separate sample for playing with PHY (where we could test all combinations, including asymetric)

Comment (KKopyscinski) [reply apps/central/src/main.c]: Now waiting is omitted and there's no PHY update, just simple connect -> disconnect -> scan procedure.

File: apps/central/src/main.c [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+    }
+    return 0;
+}
+
+static int
+scan_event(struct ble_gap_event *event, void *arg)
+{
+    /* predef_uuid stores information about UUID of device,
+       that we connect to */
+    const uint8_t predef_uuid[16] = {
```
Comment (sjanc) [inline apps/central/src/main.c:93]: why not make this ble_uuid and use stuf from ble_uuid.h to handle uuid operations?

Comment (KKopyscinski) [reply apps/central/src/main.c]: In retrospect it's much better and elegant approach.

**Pull request merged**